term-background.bash

```diff
@@ -21,80 +21,91 @@
 # Try to determine if we have dark or light terminal background
 
 typeset -i success=0
-typeset    method="xterm control"
+typeset method="xterm control"
 
 # On return, variable is_dark_bg is set
 # We follow Emacs logic (at least initially)
-get_default_bg() {
-    if [[ -n $COLORFGBG ]] ; then
-	method="COLORFGBG"
-	is_dark_colorfgbg
-    elif [[ -n $TERM ]] ; then
-	case $TERM in
-	    xterm-256color )
-		# 382.5 = (* .6 (+ 255 255 255))
-		TERMINAL_COLOR_MIDPOINT=${TERMINAL_COLOR_MIDPOINT:-383}
-		;;
-	    xterm* | dtterm | eterm* )
-		# 117963 = (* .6 (+ 65535 65535 65535))
-		TERMINAL_COLOR_MIDPOINT=${TERMINAL_COLOR_MIDPOINT:-117963}
-		is_dark_bg=0
-		;;
-
-	    * )
-		TERMINAL_COLOR_MIDPOINT=${TERMINAL_COLOR_MIDPOINT:-117963}
-		is_dark_bg=1
-		;;
-	esac
-    fi
+get_default_colorfgbg() {
+  if [[ -n $COLORFGBG ]]; then
+    method="COLORFGBG"
+    is_dark_colorfgbg
+  elif [[ -n $TERM ]]; then
+    case $TERM in
+    xterm-256color)
+      # 382.5 = (* .6 (+ 255 255 255))
+      TERMINAL_COLOR_MIDPOINT=${TERMINAL_COLOR_MIDPOINT:-383}
+      ;;
+    xterm* | dtterm | eterm*)
+      # 117963 = (* .6 (+ 65535 65535 65535))
+      TERMINAL_COLOR_MIDPOINT=${TERMINAL_COLOR_MIDPOINT:-117963}
+      is_dark_bg=0
+      ;;
+
+    *)
+      TERMINAL_COLOR_MIDPOINT=${TERMINAL_COLOR_MIDPOINT:-117963}
+      is_dark_bg=1
+      ;;
+    esac
+  fi
 }
 
 # Pass as parameters R G B values in hex
+# Compare FG to BG for light/dark, not just FB and midpoint
+# NOTE: We could have FG=#403020 BG=#203040 and tie
 # On return, variable is_dark_bg is set
 is_dark_rgb() {
-    typeset r g b
-    r=$1; g=$2; b=$3
-    if (( (16#$r + 16#$g + 16#$b) < TERMINAL_COLOR_MIDPOINT )) ; then
-	is_dark_bg=1
-    else
-	is_dark_bg=0
-    fi
+  typeset fg_r fg_g fg_b
+  typeset bg_r bg_g bg_b
+  fg_r=$1
+  fg_g=$2
+  fg_b=$3
+  bg_r=$4
+  bg_g=$5
+  bg_b=$6
+  a_fg=$((16#"$fg_r" + 16#"$fg_g" + 16#"$fg_b"))
+  a_bg=$((16#"$bg_r" + 16#"$bg_g" + 16#"$bg_b"))
+  echo "Avg test: a_fg: $a_fg a_bg: $a_bg"
```

Comment:
Did you intend to keep the "echo" in there?

Reply:
Doh!!!!